activesupport/lib/active_support/callbacks.rb

```diff
@@ -663,7 +663,9 @@ def set_callback(name, *filter_list, &block)
           if options[:if].is_a?(String) || options[:unless].is_a?(String)
             ActiveSupport::Deprecation.warn(<<-MSG.squish)
               Passing string to :if and :unless conditional options is deprecated
-              and will be removed in Rails 5.2 without replacement.
+              and will be removed in Rails 5.2 without replacement. Instead of a
+              string for eval, pass a symbol for an instance method or an array
+              of symbols.
```

Comment:
@rafaelfranca that would be better than the current message, but I think it would be better to explain what to pass in place of a string. As a deprecation message, it should ideally give enough information for someone to fix it without looking up the docs.

Maybe `Passing string to be evaluated in :if and :unless conditional options is deprecated and will be removed in Rails 5.2 without replacement. Pass a symbol for an instance method, or lamdba, proc or block, instead.`? (Array is probably the least relevant if the current implementation is a eval'd string.)